src/MathParser/Parsing/Nodes/ExpressionNode.php

```diff
@@ -96,6 +96,41 @@ function __construct($left, $operator = null, $right = null)
                 $this->associativity = self::RIGHT_ASSOC;
                 break;
 
+            case '=':
```

Comment:
IMO, `&&` should have higher precedence than `||` and `||` higher than `=`. Then `=`, `>`, `<` and other have same precedence.
So, `A = B && C` means `A = (B && C)`